com.stansassets.build/Editor/Steps/IBuildStep.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+
+namespace StansAssets.Build.Editor
+{    
+    /// <summary>
+    /// Single functional field of build process
+    /// </summary>
+    public interface IBuildStep
+    {    
+        /// <summary>
+        /// Run step execution
+        /// </summary>
+        /// <param name="buildContext">Data class with necessary parameters for build execution</param>
+        /// <param name="onComplete">Delegate which invokes when  step executing will be completed</param>
+        void Execute(BuildContext buildContext,Action<BuildStepResultArgs> onComplete = null);
```

Comment:
It's kind of nice that we have `BuildContext `  context now inside the step, but in the same time I have issues with that.
1. `BuildContext ` context is mutable so imaging if somebody will change the platform  of the `BuildContext ` during the `IBuildStep ` execute - super bad state to have.
2. If at some point we would like to add something in addition to the `BuildContext ` we do not have an ability to do that.

I would say let's keep the context in the step, but consider using our own context like `IBuildStepContext` - this way we can add stuff in it without intorusing a breaking change. Right now it for example can only  contain build target or any other info you think would make sense to have during the build step.